cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
> Why bother to have the CVMFS_GARBAGE_COLLECTION flag at all if it doesn't do anything but make it possible to use the threshold?

In the long-term we might have the `cvmfs_server gc` as a public command, to explicitly run the garbage collection instead of running it automatically after each publish. With `CVMFS_GARBAGE_COLLECTION=true` the basic functionality of garbage collection is switched on. `CVMFS_GC_TIMESTAMP_THRESHOLD` is meant to additionally enable the *automatic* garbage collection on `publish`/`snapshot`. Though, I agree that this is not obvious and propose to rename the `CVMFS_GC_TIMESTAMP_THRESHOLD` to `CVMFS_AUTO_GC_TIMESPAN`. Does that seem more clear to you?

> That default is only used on repository servers, in mkfs() and import() functions, not on stratum 1s' add-replica(). There's no default for new replicas on stratum 1s.

You are right, that is actually a flaw and I am going to fix it.